Source/WebCore/platform/graphics/egl/GLFence.cpp

```diff
@@ -31,13 +31,32 @@
 
 namespace WebCore {
 
+bool isGLFenceSyncSupported()
+{
+    static std::once_flag onceFlag;
+    static bool supported = false;
+
+    std::call_once(onceFlag, [&]() {
+        auto version = GLContext::versionFromString(reinterpret_cast<const char*>(glGetString(GL_VERSION)));
+        if (version >= 300) {
+            supported = true;
+            return;
+        }
+
+        const char* extensionsString = reinterpret_cast<const char*>(glGetString(GL_EXTENSIONS));
+        if (GLContext::isExtensionSupported(extensionsString, "GL_APPLE_sync"))
+            supported = true;
+    });
```

Comment:
Since this depends on the context, could it be that it's supported by the first context when this is called, but not the second one? or the other way around? I guess we should cache this per context...

Reply:
If you think it can't happen, don't make it more complicated. We can fix it later if we find any problem